services/network_config_manager/src/main/java/com/futurewei/alcor/netwconfigmanager/service/impl/GoalStatePersistenceServiceImpl.java

```diff
@@ -192,8 +201,10 @@ private void populateVpcResourceCache(HostGoalState hostGoalState, Map<String, I
 
                 vpcResourceMeta.setResourceMeta(portPrivateIp, portResourceMeta);
             }
-
-            vpcResourceCache.addResourceMeta(vpcResourceMeta);
+        }
+        // Commit the changes to the database. This is safe, it is wrapped by a transaction in updateGoalState
+        for(String vni : vniToVpcReourceMetaDataMap.keySet()){
+            vpcResourceCache.addResourceMeta(vniToVpcReourceMetaDataMap.get(vni));
         }
```

Comment:
It should be possible to further reduce this time with putAll, but that can wait.